OsmAnd/src/net/osmand/plus/plugins/odb/dialogs/OBDMainFragment.kt

```diff
@@ -27,10 +27,29 @@ import net.osmand.util.Algorithms
 class OBDMainFragment : OBDDevicesBaseFragment(), VehicleMetricsPlugin.ConnectionStateListener,
 	RenameOBDDialog.OnDeviceNameChangedCallback, ForgetOBDDeviceDialog.ForgetDeviceListener {
 
+	enum class OBDDataType (var widgetType: OBDTypeWidget, val icon: Int?) {
+		VIN(OBDTypeWidget.VIN, null),
+		FUEL_TYPE(OBDTypeWidget.FUEL_TYPE, R.drawable.ic_action_fuel_tank),
+		TEMPERATURE_INTAKE(OBDTypeWidget.TEMPERATURE_INTAKE, R.drawable.ic_action_obd_temperature_intake),
+		TEMPERATURE_AMBIENT(OBDTypeWidget.TEMPERATURE_AMBIENT, R.drawable.ic_action_obd_temperature_outside),
+		TEMPERATURE_COOLANT(OBDTypeWidget.TEMPERATURE_COOLANT, R.drawable.ic_action_obd_temperature_coolant),
+		ENGINE_OIL_TEMPERATURE(OBDTypeWidget.ENGINE_OIL_TEMPERATURE, R.drawable.ic_action_obd_temperature_engine_oil),
+		RPM(OBDTypeWidget.RPM, R.drawable.ic_action_obd_engine_speed),
+		SPEED(OBDTypeWidget.SPEED, R.drawable.ic_action_obd_speed),
+		FUEL_CONSUMPTION_RATE_LITER_HOUR(OBDTypeWidget.FUEL_CONSUMPTION_RATE_LITER_HOUR, R.drawable.ic_action_obd_fuel_consumption),
+		FUEL_LEFT_LITER(OBDTypeWidget.FUEL_LEFT_LITER, R.drawable.ic_action_obd_fuel_remaining),
+		CALCULATED_ENGINE_LOAD(OBDTypeWidget.CALCULATED_ENGINE_LOAD, R.drawable.ic_action_car_info),
+		FUEL_PRESSURE(OBDTypeWidget.FUEL_PRESSURE, R.drawable.ic_action_obd_fuel_pressure),
+		THROTTLE_POSITION(OBDTypeWidget.THROTTLE_POSITION, R.drawable.ic_action_obd_throttle_position),
+		BATTERY_VOLTAGE(OBDTypeWidget.BATTERY_VOLTAGE, R.drawable.ic_action_obd_battery_voltage)
+	}
+
+	data class OBDDataItem(val dataType: OBDDataType, val widget: OBDComputerWidget)
+
 	private val handler = Handler(Looper.getMainLooper())
 	private val items = mutableListOf<Any>()
 
-	private lateinit var adapter: OBDMainFragmentAdapter
+	private var adapter: OBDMainFragmentAdapter? = null
 	private var progress: View? = null
```

Comment:
Why we make it nullable?

Reply:
updateNightMode() could be called before adapter is initialized